jwst/assign_wcs/util.py

```diff
@@ -985,8 +982,6 @@ def compute_footprint_spectral(model):
     """
     swcs = model.meta.wcs
     bbox = swcs.bounding_box
-    if bbox is None:
-        bbox = wcs_bbox_from_shape(model.data.shape)
```

Comment:
I think these lines need to stay. For spectral exposure which go through extract_2d, it doesn't make sense to have a bounding_box assigned but it is needed to compute the footprint. Here the shape of the array is used as a bounding box only to compute `s_region` and is not assigned to the WCS.